File: visualize_dx_plan.R
# Set library path to include user library
.libPaths(c("~/R/library", .libPaths()))

# Load necessary libraries
suppressPackageStartupMessages({
  library(readxl)
  library(vistime)
  library(htmlwidgets)
  library(dplyr)
  library(plotly)
})

# Define file path
file_path <- "dx_plan_2025.xlsx"

# Check if file exists
if (!file.exists(file_path)) {
  stop(paste("File not found:", file_path))
}

# Read the Excel file
df <- read_excel(file_path)

# Prepare data for vistime
# Ensure dates are Date objects
df$Start_Date <- as.Date(df$Start_Date)
df$End_Date <- as.Date(df$End_Date)

# Define colors based on Status
df <- df %>%
  mutate(Color = case_when(
    grepl("Completed", Status) ~ "#28a745", # Green
    Status == "Planned" ~ "#17a2b8",         # Blue
    TRUE ~ "#6c757d"                         # Gray
  ))

# Create the Gantt chart using vistime
# We use plotly engine for interactivity
p <- vistime(df,
             col.event = "Task",
             col.start = "Start_Date",
             col.end = "End_Date",
             col.group = "Category",
             col.color = "Color",
             title = "DX Plan FY2025 Visualization",
             show_labels = TRUE)

# Save the plot as an HTML file
output_file <- "dx_plan_gantt.html"
saveWidget(p, output_file, selfcontained = TRUE)

cat(paste("Successfully created Gantt chart:", output_file, "\n"))

File: visualize_dx_plan.py
import pandas as pd
import plotly.express as px
import os

# Load data
df = pd.read_excel("dx_plan_2025.xlsx")

# Ensure date format
df['Start_Date'] = pd.to_datetime(df['Start_Date'])
df['End_Date'] = pd.to_datetime(df['End_Date'])

# Sort by Category and Start Date to group tasks visually
df = df.sort_values(by=['Category', 'Start_Date'], ascending=[True, True])

# Define colors based on Status logic similar to R script
# We can't easily map row-by-row colors in px.timeline like R's vistime without a column map.
# So we use 'Status' for color legend.
color_map = {
    "Completed": "#28a745",
    "Completed (実績)": "#28a745",
    "Planned": "#17a2b8"
}
# Fallback for others will be plotly default or we can add more

# Create Gantt chart
# We color by Status to match the R script's visual intent (Completed vs Planned)
fig = px.timeline(
    df,
    x_start="Start_Date",
    x_end="End_Date",
    y="Task",
    color="Status",
    hover_data=["Category", "Progress", "Notes"],
    title="DX Plan FY2025 Visualization",
    color_discrete_map=color_map
)

# Reverse Y axis so first task is at top
fig.update_yaxes(autorange="reversed")

# Ensure directory exists
os.makedirs("docs", exist_ok=True)

# Save
output_file = "docs/index.html"
fig.write_html(output_file)
print(f"Gantt chart saved to {output_file}")
